dcpy/models/product/dataset/metadata.py

```diff
@@ -135,16 +143,27 @@ def get_columns(self, metadata: Metadata) -> list[Column]:
         return cols
 
 
+class ZipFile(File, extra="forbid"):
+    type: Literal["Zip"] = "Zip"
+    contains: list[str]
+
+
 class Package(BaseModel, extra="forbid"):
     dataset_files: list[DatasetFile]
     attachments: list[str]
+    zip_files: list[ZipFile] = []
 
     def get_dataset(self, ds_id: str) -> DatasetFile:
         ds = [d for d in self.dataset_files if d.name == ds_id]
         if len(ds) == 1:
             return ds[0]
         raise Exception(f"No dataset named {ds_id}")
 
+    def _files_by_name(self):
+        return {ds.name: ds for ds in self.dataset_files + self.zip_files} | {
```

Comment:
Are there going to be entries in `self.zip_files` that aren't also in `dataset_files`? Or maybe am I misunderstanding zip_files a bit

Reply:
Or sorry - This is the actual zip file and not the contents of it, right? In which case, makes sense, carry on